sycl/include/sycl/multi_ptr.hpp

```diff
@@ -1232,7 +1241,7 @@ class multi_ptr<const void, Space, access::decorated::legacy> {
           _Space == Space && (Space == access::address_space::generic_space ||
                               Space == access::address_space::local_space)>>
   multi_ptr(local_accessor<ElementType, dimensions> Accessor)
-      : multi_ptr(Accessor.get_pointer()) {}
+      : m_Pointer(detail::cast_AS<decorated_type *>(Accessor.get_pointer())) {}
```

Comment:
For the legacy version you can use `pointer_t` instead of adding a new `decorated_type` alias.

Reply:
@steffenlarsen , done